### PR TITLE
added chat box into game controls form to keep sleek form styling of …

### PR DESCRIPTION
**Pull request by dlibberton**
issue requested that the chat box be on the right side of the board, I thought since there is room in the game controls form that it might be more appealing to add it to the form to use our available space, we can edit the styling of the chat box in the future with a new issue.

### CODE REVIEW

File: frontend/cheapchess/src/App.css [hunk 1]
```diff
@@ -140,7 +140,9 @@
   text-align: start;
 }
 
-
+.yellowsquare {
+  background-color: yellow;
+}
```
Comment (macleanb) [inline frontend/cheapchess/src/App.css:146]: What is the purpose of the .yellowsquare css as it relates to ChatBox addition?

File: frontend/cheapchess/src/components/ChatBox.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  const [message, setMessage] = useState('');
+
+  const handleMessageChange = (event) => {
+    setMessage(event.target.value);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (message.trim() !== '') {
+      onMessageSubmit(message);
```
Comment (macleanb) [inline frontend/cheapchess/src/components/ChatBox.js:13]: Recommend writing a stub function in the parent form named 'onMessageSubmit'...even if it just console.logs for now.  That way it won't crash when someone clicks the send button.

File: frontend/cheapchess/src/forms/Form_GameControls.js [hunk 3]
```diff
@@ -76,6 +77,10 @@ const Form_GameControls = ({
                   `}
                 </parentState.imports.Button>
               </div>
+              {/* ChatBox inserted here */}
+              <div className="d-flex justify-content-around">
+                <ChatBox />
```
Comment (macleanb) [inline frontend/cheapchess/src/forms/Form_GameControls.js:82]: Recommend passing in the onMessageSubmit button to ChatBox:

                <ChatBox parentState={{
                  ...parentState,
                  onMessageSubmit    :   onMessageSubmit
                }}/>

**Review by macleanb — CHANGES_REQUESTED** [review]
Excellent addition!  Thanks for taking the initiative and getting this going.  I added three comments / requested changes below before we merge.

File: frontend/cheapchess/src/forms/Form_GameControls.js (hunk 3 above)
Comment (macleanb) [reply frontend/cheapchess/src/forms/Form_GameControls.js:82]: In a more basic form:

            <ChatBox onMessageSubmit={ onMessageSubmit } />

File: frontend/cheapchess/src/App.css (hunk 1 above)
Comment (dlibberton) [reply frontend/cheapchess/src/App.css:146]: I was messing with highlighting the selected piece code and left that in there, I forgot to comment it out, taken care of.

File: frontend/cheapchess/src/components/ChatBox.js (hunk 2 above)
Comment (dlibberton) [reply frontend/cheapchess/src/components/ChatBox.js]: added stub function

File: frontend/cheapchess/src/forms/Form_GameControls.js (hunk 3 above)
Comment (dlibberton) [reply frontend/cheapchess/src/forms/Form_GameControls.js]: added onmessagesubmit passed to chatbox as well as username.

File: frontend/cheapchess/src/forms/Form_GameControls.js [hunk 4]
```diff
@@ -79,7 +94,10 @@ const Form_GameControls = ({
               </div>
               {/* ChatBox inserted here */}
               <div className="d-flex justify-content-around">
-                <ChatBox />
+              <ChatBox
+                  onMessageSubmit={onMessageSubmit}
+                  userName={userName}
```
Comment (macleanb) [inline frontend/cheapchess/src/forms/Form_GameControls.js:99]: userName={ parentState?.auth?.user?.first_name }

File: frontend/cheapchess/src/forms/Form_GameControls.js [hunk 5]
```diff
@@ -10,9 +10,24 @@ import ChatBox from '../components/ChatBox';
...
+  // this sets userName based on the current player or default to player1's name
+  const currentPlayerId = parentState?.gameDataFromServer?.whose_turn?.id;
+  const currentPlayer = currentPlayerId === parentState?.gameDataFromServer?.player1?.id
+    ? parentState.gameDataFromServer.player1
+    : parentState.gameDataFromServer.player2;
+
+  const userName = currentPlayer
+    ? currentPlayer.first_name
+    : parentState.gameDataFromServer.player1?.first_name;
+
```
Comment (macleanb) [inline frontend/cheapchess/src/forms/Form_GameControls.js:29]: Recommend remove lines 20-28 and replace with recommended code at line 99

**Review by macleanb — CHANGES_REQUESTED** [review]
Great addition!  Recommend the changes below.

File: frontend/cheapchess/src/forms/Form_GameControls.js (hunk 4 above)
Comment (macleanb) [reply frontend/cheapchess/src/forms/Form_GameControls.js:99]: You'd also need to update the Game.js to add auth to the props passed down to FormManager (at the bottom)

Comment (dlibberton) [reply frontend/cheapchess/src/forms/Form_GameControls.js]: all taken care of including addition of auth to parentState

File: frontend/cheapchess/src/forms/Form_GameControls.js (hunk 5 above)
Comment (dlibberton) [reply frontend/cheapchess/src/forms/Form_GameControls.js]: code removed

**Review by macleanb — APPROVED** [review]
Nice job!

**Pull request merged**